c2/cuda/groth16_cuda.cu

```diff
@@ -445,15 +505,22 @@ RustError::by_value generate_groth16_proofs_c(const Assignment<fr_t> provers[],
         if (caught_exception)
             return;
 
-        // tail MSM b_g2 - on CPU
-        for (size_t c = 0; c < num_circuits; c++) {
-            mult_pippenger<bucket_fp2_t>(results.b_g2[c],
-                tail_msm_b_g2_bases, split_vectors_b.tail_msm_scalars[c],
-                true, &groth16_pool);
+#ifndef G2_MSM_ON_GPU
+            // tail MSM b_g2 - on CPU
```

Comment:
Why indentation change? I suppose it's an artefact from removed `if`? But either way, with cross-reference to https://github.com/supranational/supra_seal/pull/19#discussion_r1285732895, I was expecting to see something like "if we split msms, then do the tail G2 msms on CPU, otherwise on GPU." Am I missing something?

Reply:
> I was expecting to see something like "if we split msms, then do the tail G2 msms on CPU, otherwise on GPU." Am I missing something?

Considering that the CPU can completely overlap G2 with other computations on the GPU, I decided to remove that distinction for now. I'm still thinking on it, but I think the best course of action would be to let the user configure whether G2 MSMs will be on the GPU or not and whether to do the same during window_post.